src/components/GameItem.js

```diff
@@ -3,26 +3,58 @@ import '../App.css';
 import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
+//wave 2
 class GameItem extends Component {
-  propTypes = {
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
+  constructor(props) {
+    super(props);
+
+    this.state = {clicked: false};
+
+    // This binding is necessary to make `this` work in the callback.
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(state => ({
+      clicked: true
+    }));
+    this.props.clickHandlers.forEach(handler => {
```

Comment:
I'm not sure where this style callback came from? It works, but it seems like it might cause you problems in future assignments.